src/utils.rs

```diff
@@ -102,80 +120,81 @@ impl<'a> InstructionTable<'a> {
         Ok(TypesTable { _table: result })
     }
 
-    fn classify(operation: &wasmparser::Operator) -> InstrumentationTypes {
+    fn classify(operation: &wasmparser::Operator, op_type: &CodillonType) -> InstrumentationFuncs {
         use wasmparser::Operator::*;
         match operation {
-            LocalSet { local_index } | LocalTee { local_index } => {
-                InstrumentationTypes::LocalSet(*local_index)
-            }
-            GlobalSet { global_index } => InstrumentationTypes::GlobalSet(*global_index),
-            I32Const { .. } => InstrumentationTypes::I32Const,
-            // Binary i32 ops: pop 2, push 1
-            I32Add | I32Sub | I32Mul | I32DivS | I32DivU | I32RemS | I32RemU | I32And | I32Or
-            | I32Xor | I32Shl | I32ShrS | I32ShrU | I32Rotl | I32Rotr | I32Eq | I32Ne | I32LtS
-            | I32LtU | I32GtS | I32GtU | I32LeS | I32LeU | I32GeS | I32GeU => {
-                InstrumentationTypes::BinaryI32
+            // Special Functions
+            LocalSet { local_index } | LocalTee { local_index } => match op_type.outputs[0] {
+                wasmparser::ValType::I32 => InstrumentationFuncs::SetLocalI32(*local_index),
+                _ => InstrumentationFuncs::Other,
+            },
+            GlobalSet { global_index } => match op_type.outputs[0] {
+                wasmparser::ValType::I32 => InstrumentationFuncs::SetGlobalI32(*global_index),
+                _ => InstrumentationFuncs::Other,
+            },
+            // Match based on inputs and outputs
+            _ => {
+                let input_types: Vec<ValType> =
+                    op_type.inputs.iter().map(|val| val.instr_type).collect();
+                if input_types.len() == 1 && op_type.outputs.is_empty() {
+                    return InstrumentationFuncs::Drop;
+                }
+                match (input_types.as_slice(), op_type.outputs.as_slice()) {
+                    ([], [wasmparser::ValType::I32]) => InstrumentationFuncs::I32Const,
+                    ([wasmparser::ValType::I32], [wasmparser::ValType::I32]) => {
+                        InstrumentationFuncs::UnaryI32
+                    }
+                    (
+                        [wasmparser::ValType::I32, wasmparser::ValType::I32],
+                        [wasmparser::ValType::I32],
+                    ) => InstrumentationFuncs::BinaryI32,
+                    _ => InstrumentationFuncs::Other,
```

Comment:
Do we actually differentiate based on input types? Like the behavior for UnaryI32 and BinaryI32 seem to be the same.